Objective-C/CBLDatabase.mm

```diff
@@ -679,12 +681,16 @@ - (nullable CBLScope*) scopeWithName: (NSString*)name error: (NSError**)error {
 
 - (nullable CBLCollection*) defaultCollection: (NSError**)error {
     CBL_LOCK(_mutex) {
+        // db closed or deleted
         if (![self mustBeOpen: error])
             return nil;
         
+        // collection removed
         BOOL isValid = _defaultCollection.isValid;
-        if (!isValid)
+        if (!isValid) {
             _defaultCollection = nil;
+            convertError({LiteCoreDomain, kC4ErrorNotFound}, error);
```

Comment:
Need to differentiate b/w null default collection and invalid default collection. If it's invalid (e.g. db closed), not open error should be returned.

Also, returning nil with CBLErrorNotFound should be mentioned in the API doc.

Reply:
line:685, it will return the db closed/deleted error.

Reply:
I am assuming the scenarios are: 
1. db is closed / deleted : in this case `_c4db` will be `nullptr`, `[self mustBeOpen]` function will return false with CBLErrorNotOpen. 
2. invalid collection: in this case collection is deleted or marked as invalid. In both cases, collection doesn't exist? So we return CBLErrorNotFound?